src/ui/components/SettingsModal/SettingsModal.tsx

```diff
@@ -0,0 +1,67 @@
+import { Box, Modal, useTheme } from '@mui/material';
+
+import { useCallback, useContext } from 'react';
+
+import {
+  AppearanceSettings,
+  SettingsContext,
+} from '@/ui/foundation/SettingsProvider/context';
+import { TabsPanel } from '../TabsPanel';
+import { AppearanceTab } from './components';
+
+interface Props {
+  open: boolean;
+  onClose: () => void;
+}
+
+/** Returns a modal to update the application settings */
+export function SettingsModal({ open, onClose }: Props) {
```

Comment:
This component only contains the modal (vs. being the modal + the button to show the modal) so the way it can be invoked remains flexible. Currently, `MainLayout` has a component to control the opening of this modal with an icon button.